File: backend/staff-service/src/main/java/com/tnl/staffservice/dto/StaffReportDTO.java
package com.tnl.staffservice.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;

import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class StaffReportDTO {
    private String id;
    private String code;
    private String firstName;
    private String lastName;
    private String dateOfBirth;
    private String joinDate;
    private String status;
    private String branch;
    private String department;
    private String position;
}

File: frontend/bidcdashboard/lib/api/dto/staff_report_dto.dart
class StaffReportDTO {
  final String id;
  final String code;
  final String firstName;
  final String lastName;
  final String dateOfBirth;
  final String joinDate;
  final String status;
  final String branch;
  final String department;
  final String position;

  StaffReportDTO({
    required this.id,
    required this.code,
    required this.firstName,
    required this.lastName,
    required this.dateOfBirth,
    required this.joinDate,
    required this.status,
    required this.branch,
    required this.department,
    required this.position,
  });

  factory StaffReportDTO.fromJson(Map<String, dynamic> json) {
    return StaffReportDTO(
      id: json['id'],
      code: json['code'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      dateOfBirth: json['dateOfBirth'],
      joinDate: json['joinDate'],
      status: json['status'],
      branch: json['branch'],
      department: json['department'],
      position: json['position'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'code': code,
        'firstName': firstName,
        'lastName': lastName,
        'dateOfBirth': dateOfBirth,
        'joinDate': joinDate,
        'status': status,
        'branch': branch,
        'department': department,
        'position': position,
      };
}
